lib/galaxy/tools/search/__init__.py

```diff
@@ -90,11 +90,11 @@ def build_index(self, tool_cache, index_help=True):
         tool_ids_to_remove = (indexed_tool_ids - set(tool_cache._tool_paths_by_id.keys())).union(tool_cache._removed_tool_ids)
         for indexed_tool_id in indexed_tool_ids:
             indexed_tool = tool_cache.get_tool_by_id(indexed_tool_id)
-            if not indexed_tool:
-                tool_ids_to_remove.add(indexed_tool_id)
-                continue
-            if not indexed_tool.is_latest_version and not indexed_tool.latest_version.hidden:
-                tool_ids_to_remove.add(indexed_tool_id)
+            if indexed_tool:
+                latest_version = indexed_tool.latest_version
+                if latest_version and latest_version.hidden:
```

Comment:
Not sure if it's on purpose, but this seems functionally different from before, where we didn't remove an old version if the latest is hidden, xref. https://github.com/galaxyproject/galaxy/pull/10175#issuecomment-685001855

Reply:
Are you sure ? My brain isn't working super-well at this hour, but I think this is equivalent, i've just inverted the continue.

Reply:
The inverted condition (without the fix) would be:
```python
if indexed_tool.is_latest_version or indexed_tool.latest_version.hidden:
    continue
```

See my suggestion above.